lightningd/signmessage.c

```diff
@@ -56,8 +56,7 @@ static const u8 *from_zbase32(const tal_t *ctx, const char *msg)
 	if (!bech32_convert_bits(u8arr, &len, 8,
 				 u5arr, tal_bytelen(u5arr), 5, false))
 		return tal_free(u8arr);
-	assert(len == tal_bytelen(u8arr));
-	return u8arr;
+	return len == tal_bytelen(u8arr) ? u8arr : NULL;
```

Comment:
I think this ends up leaking `u8arr` in case the length doesn't match. Better free it with `tal_free` here.

Reply:
Thanks for this catch, I added it right now and free the array in the case of the invalid length!